core/include/detray/geometry/surface_base.hpp

```diff
@@ -60,49 +61,60 @@ class surface_base {
 
     ~surface_base() = default;
     surface_base(const surface_base &lhs) = default;
-    surface_base() = delete;
+    surface_base() = default;
```

Comment:
Let me defend myself before someone complains here. Lots of link type variables in `surface_base` and mask objects have templated types so it is quite difficult to assign proper default values in the constructor.

Reply:
Actually I suspect such templated types for linkers shouldn't be very necessary...